appengine/appengine.bzl

```diff
@@ -69,12 +69,11 @@ jar_filetype = FileType([".jar"])
 def _add_file(in_file, output, path = None):
   output_path = output
   input_path = in_file.path
-  if path and input_path.startswith(path):
-    output_path += input_path[len(path):]
+  if path and in_file.short_path.startswith(path):
```

Comment:
I think this still breaks if an appengine_war is loaded from an external repository, because the data_path attribute will no longer be relative to the root of the workspace.